### ISSUE TITLE
HLint updated, and Drasil has new suggested changes

### ISSUE THREAD

**Issue by balacij** [JacquesCarette/Drasil#2968]
HLint is causing problems for this, but it looks like a separate issue.

_Originally posted by @balacij in https://github.com/JacquesCarette/Drasil/issues/2966#issuecomment-1110004274_

**Comment by balacij** [comment]
We only get one hint:
```console
[nix-shell:~/Programming/Drasil/code]$ ~/.cabal/bin/hlint ./drasil-code/
drasil-code/test/HelloWorld.hs:90:18-39: Warning: Evaluate
Found:
  const "myConst" string
Perhaps:
  "myConst"

1 hint
```
But I'm fairly certain it's not what we want here.

The line in question defines a single line in a test program:
```haskell
    constDecDef (const "myConst" string) (litString "Imconstant"),
```
I'm guessing it's because it's incorrectly recognizing `const` as the Prelude.const instead of GOOL's `const` typeclass function.

**Comment by balacij** [comment]
I was going to submit a bug report to them, but then I saw their [Bugs and Limitations](https://github.com/ndmitchell/hlint#bugs-and-limitations=) section in their README. Specifically, it looks like HLint ignores which names are in scope; so I'm guessing "const" isn't checked to make sure that it's from Prelude. I imagine the "Evaluation" hint will work well for us anywhere that we don't manually write GOOL code, which is the majority. If we run into these more often, then it looks straightforward to disable the hint globally or on a per-file basis through the `.hlint.yaml`. For now, I've only disabled it on the single function that contains the "problematic" line in #2969.
